server/src/main/java/org/elasticsearch/cluster/metadata/DataStream.java

```diff
@@ -1462,8 +1463,11 @@ private Builder(DataStreamIndices dataStreamIndices) {
                 this.autoShardingEvent = dataStreamIndices.autoShardingEvent;
             }
 
+            /**
+             * Set the list of indices. We always create an immutable copy for performance reasons.
```

Comment:
I think here the performance reasons is a bit confusing. The way I read it, it implies that we create an immutable list for performance reasons. I think the correct phrasing is, that we create an immutable copy because the constructor expects an immutable list. Right?

Reply:
Ah my bad I think I misinterpreted your initial comment. Will update, thanks